server/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -197,9 +198,13 @@ protected GeoShapeQueryBuilder newShapeQueryBuilder(String fieldName, Supplier<G
 
     @Override
     public Query buildShapeQuery(QueryShardContext context, MappedFieldType fieldType) {
-        if (fieldType.typeName().equals(GeoShapeFieldMapper.CONTENT_TYPE) == false) {
+        if (fieldType.typeName().equals(GeoShapeFieldMapper.CONTENT_TYPE) == false &&
+            fieldType.typeName().equals(GeoPointFieldMapper.CONTENT_TYPE) == false) {
             throw new QueryShardException(context,
-                "Field [" + fieldName + "] is not of type [" + queryFieldType() + "] but of type [" + fieldType.typeName() + "]");
+                "Field [" + fieldName + "] is of unsupported type [" + fieldType.typeName() + "]. ["
+                + NAME + "] query supports the following types ["
+                + GeoShapeFieldMapper.CONTENT_TYPE + ", "
+                + GeoPointFieldMapper.CONTENT_TYPE + "]");
```

Comment:
We cannot hardcode those values. This method is called for geo_shape and shape queries, therefore we use queryFieldType() to get the underlaying fields. I think the solution here is to change the signature of queryFieldType() method to return an arrays of fields.

Reply:
Agree, there is already List validContentTypes() so I have encapsulated the supported types there to avoid doing anything drastic with Reflection. queryFieldType is essentially redundant, so I changed the signature to List, pointed this to ValidContentTypes() and then deprecated.